coderd/workspaceagentsrpc_internal_test.go

```diff
@@ -0,0 +1,376 @@
+package coderd
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/coder/coder/v2/coderd/util/ptr"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"nhooyr.io/websocket"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbmock"
+	"github.com/coder/coder/v2/coderd/database/dbtime"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestAgentWebsocketMinder_ContextCancel(t *testing.T) {
+	t.Parallel()
+	ctx := testutil.Context(t, testutil.WaitShort)
+	now := dbtime.Now()
+	fConn := &fakePingerCloser{}
+	ctrl := gomock.NewController(t)
+	mDB := dbmock.NewMockStore(ctrl)
+	fUpdater := &fakeUpdater{}
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+	agent := database.WorkspaceAgent{
+		ID: uuid.New(),
+		FirstConnectedAt: sql.NullTime{
+			Time:  now.Add(-time.Minute),
+			Valid: true,
+		},
+	}
+	build := database.WorkspaceBuild{
+		ID:          uuid.New(),
+		WorkspaceID: uuid.New(),
+	}
+	replicaID := uuid.New()
+
+	uut := &agentWebsocketMinder{
```

Comment:
What does uut mean?

Reply:
unit under test - it's a convention I like to use in tests to make it obvious what thing is being tested

Reply:
Ah, I see. That’s new to me. Not sure whether I like it or not yet, but I’ll let it mellow a bit.